webui/src/pages/repositories/repository/settings/branches.jsx

```diff
@@ -0,0 +1,138 @@
+import {useAPI} from "../../../../lib/hooks/api";
+import {branchProtectionRules} from "../../../../lib/api";
+import Modal from "react-bootstrap/Modal";
+import Form from "react-bootstrap/Form";
+
+const SettingsContainer = () => {
+    const {repo, loading, error} = useRefs();
+    const [showCreateModal, setShowCreateModal] = useState(false);
+    const [refresh, setRefresh] = useState(false);
+    const [actionError, setActionError] = useState(null);
+    const [deleteButtonDisabled, setDeleteButtonDisabled] = useState(false)
+
+    const {response: rules, error: rulesError, loading: rulesLoading} = useAPI(async () => {
+        return branchProtectionRules.getRules(repo.id)
+    }, [repo, refresh])
+    if (!!error) return <Error error={error}/>;
+    if (!!rulesError) return <Error error={rulesError}/>;
+    if (!!actionError) return <Error error={actionError}/>;
+    return (<>
+        <div className="mt-3 mb-5">
+            <div className={"section-title"}>
+                <h4 className={"mb-0"}>
+                    <div className={"ml-1 mr-1 pl-0 row flex"}>
+                        <div className={"flex-fill"}>Branch protection rules</div>
+                        <RefreshButton className={"ml-1"} onClick={() => {setRefresh(!refresh)}}/>
+                        <Button className={"ml-2"} onClick={() => setShowCreateModal(true)}>Add</Button>
+                    </div>
+                </h4>
+            </div>
+            <div>
+                Define branch protection rules to prevent direct changes.
+                Changes to protected branches can only be done by merging from other branches.
+            </div>
+            {loading || rulesLoading ? <div className={"mt-3 ml-1 pr-5"}><Loading/></div> :
+                <div className={"row mt-3 ml-1 pr-5"}>
+                    <Card className={"w-100 rounded border-0"}>
+                        <Card.Body className={"p-0 rounded"}>
+                            <ListGroup>
+                                {rules && rules.map(r => {
+                                    return <ListGroup.Item >
+                                        <div className={"d-flex"}>
+                                            <code>{r.pattern}</code>
+                                            <Button disabled={deleteButtonDisabled} className={"ml-auto"} size={"sm"} variant="secondary" onClick={() => {
+                                                setDeleteButtonDisabled(true)
+                                                branchProtectionRules.deleteRule(repo.id, r.pattern).then(() => {
+                                                    setRefresh(!refresh)
+                                                    setDeleteButtonDisabled(false)
+                                                }).catch(err => {
+                                                    setDeleteButtonDisabled(false)
+                                                    setActionError(err)
```

Comment:
Error location is better located above the buttons imo
![image](https://user-images.githubusercontent.com/9861474/135491868-bd480c99-5c46-4bc2-a8d0-a0fbe7240554.png)

Reply:
And error persist when clicking `Cancel` then `Add` again, without performing the action again

Reply:
Fixed